zigpy/endpoint.py

```diff
@@ -90,6 +90,11 @@ async def initialize(self) -> None:
 
         self.status = Status.ZDO_INIT
 
+    @property
+    def clusters(self) -> list[zigpy.zcl.Cluster]:
+        """Return all clusters on this endpoint."""
+        return list(self.in_clusters.values()) + list(self.out_clusters.values())
```

Comment:
Might not matter a lot, but since this is called every time the attribute cache is saved, this might be a bit faster(?)
```suggestion
        return [*self.in_clusters.values(), *self.out_clusters.values()]
```

Reply:
`_save_attribute_cache` is called when a device object is initially committed into the database. Attribute updates go through `_save_attribute`, which shouldn't do anything too expensive. But I do like this syntax, I didn't know you could use it for lists!

Reply:
> `_save_attribute_cache` is called when a device object is initially committed into the database, after initialization. Attribute updates go through `_save_attribute`

Ah, of course! 😄